Authorization/Processor/Ipg.ts

```diff
@@ -0,0 +1,89 @@
+import { Base } from "./Base"
+
+export interface Ipg {
+	ipgTransactionId: string
+	transactionType:
+		| "SALE"
+		| "PREAUTH"
+		| "CREDIT"
+		| "FORCED_TICKET"
+		| "VOID"
+		| "RETURN"
+		| "POSTAUTH"
+		| "PAYER_AUTH"
+		| "DISBURSEMENT"
+	merchantTransactionId: string
+	transactionStatus: "APPROVED" | "WAITING" | "VALIDATION_FAILED" | "PROCESSING_FAILED" | "DECLINED"
+	schemeTransactionId?: string
+	processor: {
+		referenceNumber: string
+		authorizationCode: string
+		responseCode: string
+		responseMessage: string
+		avsResponse: {
+			streetMatch: "Y" | "N" | "NO_INPUT_DATA" | "NOT_CHECKED"
+			postalCodeMatch: "Y" | "N" | "NO_INPUT_DATA" | "NOT_CHECKED"
+		}
+		securityCodeResponse: "MATCHED" | "NOT_MATCHED" | "NOT_PROCESSED" | "NOT_PRESENT" | "NOT_CERTIFIED" | "NOT_CHECKED"
+	}
+	transactionOrigin: "ECOM" | "MAIL" | "PHONE" | "RETAIL"
+	terminalId: string
+}
+
+export namespace Ipg {
+	export function is(value: Ipg | any): value is Ipg {
+		return (
+			typeof value == "object" &&
+			value &&
+			typeof value.ipgTransactionId == "string" &&
+			[
+				"SALE",
+				"PREAUTH",
+				"CREDIT",
+				"FORCED_TICKET",
+				"VOID",
+				"RETURN",
+				"POSTAUTH",
+				"PAYER_AUTH",
+				"DISBURSEMENT",
+			].includes(value.transactionType) &&
+			typeof value.merchantTransactionId == "string" &&
+			["APPROVED", "WAITING", "VALIDATION_FAILED", "PROCESSING_FAILED", "DECLINED"].includes(value.transactionStatus) &&
+			(value.schemeTransactionId == undefined || typeof value.schemeTransactionId == "string") &&
+			typeof value.processor == "object" &&
+			value.processor &&
+			typeof value.processor.referenceNumber == "string" &&
+			typeof value.processor.authorizationCode == "string" &&
+			typeof value.processor.responseCode == "string" &&
+			typeof value.processor.responseMessage == "string" &&
+			typeof value.processor.avsResponse == "object" &&
+			value.processor.avsResponse &&
+			["Y", "N", "NO_INPUT_DATA", "NOT_CHECKED"].includes(value.processor.avsResponse.streetMatch) &&
+			["Y", "N", "NO_INPUT_DATA", "NOT_CHECKED"].includes(value.processor.avsResponse.postalCodeMatch) &&
+			["MATCHED", "NOT_MATCHED", "NOT_PROCESSED", "NOT_PRESENT", "NOT_CERTIFIED", "NOT_CHECKED"].includes(
+				value.processor.securityCodeResponse
+			) &&
+			["ECOM", "MAIL", "PHONE", "RETAIL"].includes(value.transactionOrigin) &&
+			typeof value.terminalId == "string"
+		)
+	}
+
+	export function from(value: Ipg): Base {
+		return {
+			name: "ipg",
+			mid: value.terminalId,
+			entry: value.transactionOrigin == "ECOM" ? "e-commerce" : value.transactionOrigin,
+			reference: {
+				transaction: value.ipgTransactionId,
+				retrieval: value.processor.referenceNumber,
+				scheme: value.schemeTransactionId ?? "",
+			},
+			response: {
+				code: value.processor.responseMessage,
```

Comment:
```suggestion
				code: value.processor.responseCode,
```
There are apparently to separate `Issuer Response...`, I was also mixing them up
![image](https://user-images.githubusercontent.com/14332757/165463417-2b34d364-346c-4371-89a0-b51548035326.png)